### DISCUSSION TITLE
cdo error when running Prepare_ERA5-Land_Forcing.sh to extract data from PREVIOUS_FILE

### DISCUSSION THREAD

**Discussion by chenghaow** [NCAR/hrldas#192] [Q&A] [answered]
Hi, I am doing a test CONUS run following the HRLDAS tutorial Note5 ([https://github.com/NCAR/hrldas/blob/master/tutorial/Note5_Regional_modeling_ERA5-Land_forcing.ipynb](url)). My 3-day run starts on 2010/01/01 00UTC. I have downloaded the following two ERA5-Land files: **ERA5_global_20100101.grib** (3-day forcing data) and **ERA5_global_20091231.grib** (data for 2009/12/31 23:00). I've also modified the following lines accordingly:

```
# set date for data processing
inyear=2010
inmonth=01
inday=01
preyear=2009
premonth=12
preday=31

indate=${inyear}${inmonth}${inday}
INFILE=/path-to-forcing-data/ERA5_Land/raw/ERA5_global_${indate}.grib
predate=${preyear}${premonth}${preday}
PREVIOUS_FILE=/path-to-forcing-data/ERA5_Land/raw/ERA5_global_${predate}.grib
```

The shell code successfully completed the first part (extraction of time-step variables), but failed when extracting the three accumulated variables (ACSSRD, ACSTRD, and ACTP). The error message is:

```
 o-----------------> PART 2: DECUMULATE FIELDS
cdo(1) seldate (Abort): Float parameter >20091231T23:00:00< contains invalid character at position 9!
cdo(1) seldate (Abort): Float parameter >20091231T23:00:00< contains invalid character at position 9!
cdo(1) seldate (Abort): Float parameter >20091231T23:00:00< contains invalid character at position 9!
```
I have double-checked the downloaded grib file and ensured it is complete. Any thoughts on this? @CharlesZheZhang @cenlinhe Thanks!

**Comment by CharlesZheZhang** [comment, marked as answer]
Hi, Thanks for providing the details.
This issue is related to the naming format of the indate and predate that cdo seldate doesn't recognize.
Please try naming the INFILE and PREVIOUS_FILE with:
indate=${inyear}-${inmonth}-${inday}
predate=${preyear}-${premonth}-${preday}

This should fix the issue

**Reply by chenghaow** [reply]
Got it. Thank you!
